File: 2022/template.py
#!/usr/bin/env python3

from py_shared import *
from dataclasses import dataclass

DAY = %day%
PART = %part%;

# Types ────────────────────────────────────────────────────────────────────── #

@dataclass
class xxx:
	pass

# Solution ─────────────────────────────────────────────────────────────────── #

def solve(lines):
	return sum([solve_line(line) for line in lines])

def solve_line(line):
	return 0

# Main ─────────────────────────────────────────────────────────────────────── #

run_test(DAY, PART, solve)
# run_actual(DAY, PART, solve)

File: 2023/template.rs
mod rust_advent;
use rust_advent::*;
mod shared;
use shared::*;

const DAY: usize = %day%;
const PART: usize = %part%;

// Solution ───────────────────────────────────────────────────────────────── //

fn solve(lines: Vec<String>) -> isize {
	lines.iter().map(|line| solve_line(line)).sum()
}

fn solve_line(line: &str) -> isize {
	0
}

// Main ───────────────────────────────────────────────────────────────────── //

fn main() {

	nl();
	match has_argument("test") or has_argument("t") {
		true => run_test(DAY, PART, solve);
		false => run_actual(DAY, PART, solve);
	}
}